portfolio/src/main/webapp/src/__common.js

```diff
@@ -132,6 +157,63 @@ function createButtonLink(buttonLabel, buttonLink) {
   return buttonElement;
 }
 
+/**
+ * Gets the user's login status, username, and
+ * login API url from the server.
+ *
+ * @return {Promise<any>}
+ */
+function getLoginStatus() {
+  return fetch('/login').then((response) => {
+    if (response.ok) {
+      return response.json();
+    } else {
```

Comment:
Since you `return` in the first `if`, you don't need to use the `else` (and therefore can save a level of indentation).

Reply:
What would you think about the following code snippet? -- basically your suggestion but in reverse
```javascript
if (!response.ok) {
  return Promise.reject(
    new Error(response.status + ': ' + response.statusText);
}
return response.json();
```
It's similar to what you suggested, except the code 'short-circuits' on the exceptional condition that the server returns a failure code rather than the normal condition that the server returns a success code (and response data).